src/DefinitionResolver.php

```diff
@@ -175,10 +177,15 @@ private function getDocBlock(Node $node)
      *
      * @param Node $node
      * @param string $fqn
+     * @param Scope|null $scope Scope at the point of Node. If not provided, will be computed from $node.
```

Comment:
How do you ensure that the scope passed in is the scope of the node?

Reply:
I have to think about this a bit more.

One thing that comes to mind is that the `Scope` could contain information about the `Node` or `FunctionLike` it is defined at? The added check would have a performance penalty (which I guess might be significant), since you'd need to find the closest `FunctionLike`, `ClassLike`, or `SourceFileNode` ancestor every time you wanted to check if the scope belongs to a node.